### DISCUSSION TITLE
Add taint toleration to cluster-autoscaler

### DISCUSSION THREAD

**Discussion by mabilinab** [mysticaltech/terraform-hcloud-kube-hetzner#1727] [Ideas]
### Description

I have added a taint to my master node, and now, the cluster-autoscaler pods can no longer be scheduled since they do not have the right taint toleration.
Is it possible to configure the cluster-autoscaler deployment to have the right toleration to be scheduled on the master node?
(I think I can do that through kubectl, but that will be overridden whenever I apply the kube.tf)

**Comment by mysticaltech** [comment]
@mabilinab  It should not get overwritten as the tf state was not changed. Try it, you'll see that it remains.

**Comment by vlassisemm** [comment]
Hello,

I have tried this solution, it works but when changes are made on the autoscaler configuration, it redeploys and the toleration is lost.

A variable setting in kube.tf about autoscaler tolerations would greatly help!

**Comment by mysticaltech** [comment]
@vlassisemm  Maybe

**Comment by mysticaltech** [comment]
Implemented in PR: https://github.com/mysticaltech/terraform-hcloud-kube-hetzner/pull/2073

Closing this discussion as addressed for the v3 workstream.
